Sources/Core/AWSClientRuntime/Auth/AuthSchemes/SigV4Util.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import ClientRuntime
+
+public class SigV4Util {
+    static let usesSignedBodyHeader = ["S3", "Glacier", "S3 Control"]
+    static let forceUnsignedBodyForPresigningURL = [
+        "S3": ["getObject", "putObject"]
+    ]
+
+    static func shouldForceUnsignedBody(flow: FlowType, serviceName: String, opName: String) -> Bool {
+        let serviceQualifies = forceUnsignedBodyForPresigningURL.keys.contains(serviceName)
+        let flowQualifies = flow == .PRESIGN_URL
+        return serviceQualifies && flowQualifies && forceUnsignedBodyForPresigningURL[serviceName]!.contains(opName)
```

Comment:
Nit: Can be made more readable by not using variables. Ex. `flowQualifies` is less readable than `flow == .PRESIGN_URL` (IMO)
```
return flow == .PRESIGN_URL && 
      forceUnsignedBodyForPresigningURL[serviceName]?.contains(operationName) ?? false
```

Reply:
Changed as suggested.